### PR TITLE
[Gallery] Crane focus

### PR DESCRIPTION
**Pull request by guidezpl**
This PR implements focus (to the best of my ability) for Crane. All elements are accessible via the tab key, however, there is an unsolved ordering problem which causes the destinations to be navigated through before the form fields. 

Closes https://github.com/material-components/material-components-flutter-gallery/issues/324

### CODE REVIEW

File: gallery/gallery/lib/studies/crane/backlayer.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      policy: WidgetOrderFocusTraversalPolicy(),
+      child: IndexedStack(
+        index: tabIndex,
+        children: widget.backLayerItems
+            .map((backLayerItem) => Focus(
+                  canRequestFocus: backLayerItem.index == tabIndex,
+                  debugLabel: 'backLayerItem: $backLayerItem',
+                  child: backLayerItem,
+                ))
+            .toList(),
```
Comment (perclasson) [inline gallery/gallery/lib/studies/crane/backlayer.dart]: nit: I would change this to
```dart
[
for (BackLayerItem backLayerItem in widget.backLayerItems)
    Focus(
    ...
    )
]
```

File: gallery/gallery/lib/studies/crane/item_cards.dart [hunk 2]
```diff
@@ -33,8 +34,13 @@ class _ItemCardsState extends State<ItemCards> {
 
     return destinations
         .map(
-          (d) => RepaintBoundary(
-            child: _DestinationCard(destination: d),
+          (d) => HighlightFocus(
+            debugLabel: 'DestinationCard: ${d.destination}',
+            highlightColor: Colors.red.withOpacity(0.5),
+            onPressed: () => print('pressed'),
```
Comment (perclasson) [inline gallery/gallery/lib/studies/crane/item_cards.dart]: Should the pressed here be removed? Or why do we have it?

File: gallery/gallery/lib/layout/highlight_focus.dart [hunk 3]
```diff
@@ -67,6 +70,7 @@ class _HighlightFocusState extends State<HighlightFocus> {
 
     return Focus(
       canRequestFocus: widget.hasFocus,
+      debugLabel: widget.debugLabel,
```
Comment (rami-a) [inline gallery/gallery/lib/layout/highlight_focus.dart]: What is the debug label for?

File: gallery/gallery/lib/studies/crane/header_form.dart [hunk 4]
```diff
@@ -75,7 +75,7 @@ class _HeaderTextField extends StatelessWidget {
         fillColor: cranePurple700,
         filled: true,
         hintText: field.title,
-        hasFloatingPlaceholder: false,
+        floatingLabelBehavior: FloatingLabelBehavior.never,
```
Comment (rami-a) [inline gallery/gallery/lib/studies/crane/header_form.dart]: What is this change for?

File: gallery/gallery/lib/layout/highlight_focus.dart (hunk 3 above)
Comment (guidezpl) [reply gallery/gallery/lib/layout/highlight_focus.dart]: It identifies nodes in the output of debugDumpFocusTree()

File: gallery/gallery/lib/studies/crane/header_form.dart (hunk 4 above)
Comment (guidezpl) [reply gallery/gallery/lib/studies/crane/header_form.dart]: Removes a deprecated property. It's not in stable yet so I reverted it

File: gallery/gallery/lib/studies/crane/item_cards.dart (hunk 2 above)
Comment (guidezpl) [reply gallery/gallery/lib/studies/crane/item_cards.dart]: Yes it should.

**Pull request merged**